tests/unit/cli/builder/test_cli_builder.py

```diff
@@ -0,0 +1,420 @@
+import mmcv
+import pytest
+from mmcv.utils import Registry
+from torch import nn
+
+from otx.cli.builder.builder import (
+    Builder,
+    get_backbone_out_channels,
+    update_backbone_args,
+    update_channels,
+)
+from otx.cli.utils.importing import get_otx_root_path
+from otx.mpa.utils.config_utils import MPAConfig
+from tests.test_suite.e2e_test_system import e2e_pytest_unit
+
+
+class TestOTXCLIBuilder:
+    """Check Builder's function is working well.
+
+    1. Check "Builder.build_task_config" function that create otx-workspace is working well.
+    <Steps>
+        1. Create Classification custom workspace
+        2. Raising Error of building workspace with already created path
+        3. Update hparam.yaml with train_type="selfsl"
+        4. Raising ValueError with wrong train_type
+        5. Build workspace with model_type argments
+        6. Raise ValueError when build workspace with wrong model_type argments
+
+    2. Check "Builder.build_backbone_config" function that generate backbone configuration file is working well
+    <Steps>
+        1. Generate backbone config file (mmcls.MMOVBackbone)
+        2. Raise ValueError with wrong output_path
+
+    3. Check "Builder.build_model_config" function that update model config with new backbone is working well
+    <Steps>
+        1. Update model config with mmcls.ResNet backbone (default model.backbone: otx.OTXEfficientNet)
+        2. Raise ValueError with wrong model_config_path
+        3. Raise ValueError with wrong backbone_config_path
+        4. Update model config without backbone's out_indices
+        5. Update model config with backbone's pretrained path
+    """
+
+    @pytest.fixture(autouse=True)
+    def setup(self, tmp_dir_path):
+        self.otx_builder = Builder()
+        self.otx_root = get_otx_root_path()
+        self.tmp_dir_path = tmp_dir_path
+
+    @e2e_pytest_unit
+    def test_builder_build_task_config_create_workspace(self):
+        """Create Classification custom workspace."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_task_config_create_workspace")
+        inputs = {"task_type": "classification", "workspace_path": workspace_path, "otx_root": self.otx_root}
+        self.otx_builder.build_task_config(**inputs)
+        assert os.path.exists(workspace_path)
+        assert os.path.exists(os.path.join(workspace_path, "configuration.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "template.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "model.py"))
+        assert os.path.exists(os.path.join(workspace_path, "data.yaml"))
+        assert os.path.exists(os.path.join(workspace_path, "data_pipeline.py"))
+
+    @e2e_pytest_unit
+    def test_builder_build_task_config_reuse_same_path(self):
+        """Raising Error of building workspace with already created path."""
+        workspace_path = os.path.join(self.tmp_dir_path, "test_builder_build_task_config_create_workspace")
```

Comment:
Would it be possible to switch to pathlib here too? Or does `build_task_config` always expect a `str`?

Reply:
Modified to use `str` and `pathlib.Path` by replacing `os.path` in `builder.py` including `build_task_config` with `pathlib.Path`.